modules/components/org.wso2.carbon.appfactory.provisioning.runtime/src/main/java/org/wso2/carbon/appfactory/provisioning/runtime/KubernetesRuntimeProvisioningService.java

```diff
@@ -342,20 +342,207 @@ public DeploymentLogs getRuntimeLogs(DeploymentConfig deploymentConfig, Query qu
+                        .withNamespace(namespace.getMetadata().getName()).withName(runtimeProperty.getName())
+                        .endMetadata().withData(runtimeProperty.getProperties()).build();
+
+                kubernetesClient.secrets().create(secret);
+
+                Volume volume = new VolumeBuilder()
+                        .withName(KubernetesPovisioningConstants.VOLUME_MOUNT)
+                        .withNewSecret()
+                        .withSecretName(runtimeProperty.getName())
+                        .endSecret()
+                        .build();
+
+                secrets.add(volume);
+
+                break;
+            case ENVIRONMENT:
+                if (log.isDebugEnabled()) {
+                    String message = "Creating property type environment for the application:"
+                            + applicationContext.getName() + " for the tenant domain:"
+                            + applicationContext.getTenantInfo().getTenantDomain();
+                    log.debug(message);
+                }
+                for (Map.Entry<String, String> property : runtimeProperty.getProperties().entrySet()) {
+                    EnvVar envVar = new EnvVarBuilder()
+                            .withName(property.getKey())
+                            .withValue(property.getValue())
+                            .build();
+
+                    environments.add(envVar);
+                }
+
+                break;
+            default:
+                String message = "Runtime property type is not support, property type:"
+                        + runtimeProperty.getPropertyType();
+
+                throw new IllegalArgumentException(message);
+
+            }
+        }
+
+        container.setEnvVariables(environments);
+        deploymentConfig.setContainer(container);
+        deploymentConfig.setSecrets(secrets);
+
+        //Calling application method to deploy the runtime properties with the application
+        deployApplication(deploymentConfig);
+
     }
 
+    /**
+     * Update runtime properties already defined in the application
+     *
+     * @param runtimeProperties list of runtime properties
+     * @param deploymentConfig  includes deployment related details
+     * @throws RuntimeProvisioningException
+     */
     @Override
-    public void updateRuntimeProperties(RuntimeProperty runtimeProperty) throws RuntimeProvisioningException {
+    public void updateRuntimeProperties(List<RuntimeProperty> runtimeProperties, DeploymentConfig deploymentConfig)
+            throws RuntimeProvisioningException {
 
+        List secrets = new ArrayList();
+        Container container = new Container();
+        List environments = new ArrayList();
+
+        for (RuntimeProperty runtimeProperty : runtimeProperties) {
+            switch (runtimeProperty.getPropertyType()) {
+            case SENSITIVE:
+                if (log.isDebugEnabled()) {
```

Comment:
Initially, we did not use a database to store the run-time properties.  Now we can remove this updateRuntimeProperties method, and for update run-time properties, we can use same setRuntimeProperties method.